src/control/lib/control/pool_test.go

```diff
@@ -507,6 +507,68 @@ func TestControl_PoolCreate(t *testing.T) {
 	}
 }
 
+func TestControl_UpdatePoolQueryState(t *testing.T) {
+	for name, tc := range map[string]struct {
+		pqr      *PoolQueryResp
+		expState string
+	}{
+		"Pool state as Ready": {
+			pqr: &PoolQueryResp{
+				Status: 0,
+				UUID:   "foo",
+				State:  "",
+				PoolInfo: PoolInfo{
+					TotalTargets:    1,
+					DisabledTargets: 0,
+				},
+			},
+			expState: "Ready",
+		},
+		"Pool state as Degraded": {
+			pqr: &PoolQueryResp{
+				Status: 0,
+				UUID:   "foo",
+				State:  "Ready",
```

Comment:
I find this a little awkward overwriting the pool state based on whether any targets have been disabled, I realise we have already discussed this and accepted it as a solution but I would be more comfortable with adding another field to indicate we are calculating another state based on whether targets are disabled e.g. TargetsDisabled or simply compare TotalTargets and DisabledTargets when printing the QueryResponse in the pretty package. Then there is less possibility of problems later on due to overwriting the state returned from PoolQuery.